docs/design_docs/user_facing_api.md

```diff
@@ -0,0 +1,24 @@
+# Proposal for user facing API
+
+## Objectives
+- deploy the plugins without day 1 configuration
+- dynamic day 2 host resource exposure
```

Comment:
I would add more objectives to say what we want to be able to configure:

- Expose underlying interface as a logical resource (don't know how to describe it better). Underlying NIC should not be exposed under the interfaces name. Cluster can be heterogenous and the same network may be exposed by different names on various hosts.
- Be able to use nodeSelector, nodeAffinity, NIC name, additional NIC selectors...

The last two would help us model the API, keep it flexible.

Reply:
Added these two; by NIC name I think you meant resource name - since there is also the concept of NIC selectors ... 

Let me know if I screwed up.